lib/util/Components.js

```diff
@@ -63,14 +63,50 @@ Components.prototype.get = function(node) {
  * @param {Object} props Additional properties to add to the component.
  */
 Components.prototype.set = function(node, props) {
+  const popsAreEquivalent = (propA, propB) => {
+    if (propA.name === propB.name) {
+      if (!propA.allNames && !propB.allNames) {
+        return true;
+      } else if (propA.allNames.join('') === propB.allNames.join('')) {
+        return true;
+      }
+      return false;
+    }
+    return false;
+  };
```

Comment:
I personally find this nested if if else if quite hard to read. It can just be:

```js
return (
  propsA.name === propsB.name &&
  ((!propA.allNames && !propB.allNames) || (propA.allNames.join('') === propB.allNames.join(''))
)
```

Although it's not that simple either since it still has a lot of conditions, I find it a bit easier...: `conditionA && ((conditionD && conditionC) || conditionE)`